src/main/java/com/EighthLight/app/Ai.java

```diff
@@ -45,66 +44,70 @@ public int findBestMove(Map<Integer, Integer>scoredSpaces)  {
         return bestSpace;
     }
 
-    public int miniMax(ArrayList boardState, int depth, String playerWhoMadeLastMove) {
-        depth++;
+
+    public int miniMax(ArrayList boardState, String playerWhoMadeLastMove) {
         ArrayList newBoard = new ArrayList();
         newBoard.addAll(boardState);
         ArrayList<Integer> emptySpaces = findEmptySpaces(newBoard);
         String playerWhoIsMoving = playerWhoMadeLastMove.equals(aiSymbol) ? humanSymbol : aiSymbol;
         ArrayList<Integer> scores = new ArrayList<>();
 
-        if (playerWhoMadeLastMove.equals(humanSymbol) && (ourBoard.hasAPlayerWon(newBoard, playerWhoMadeLastMove))) {
-            return -10 + depth;
-        } else if (playerWhoMadeLastMove.equals(aiSymbol) && (ourBoard.hasAPlayerWon(newBoard, playerWhoMadeLastMove))) {
-            return 10 - depth;
+        if (ourBoard.hasAPlayerWon(newBoard, playerWhoIsMoving)) {
+            return -10;
+        } else if (ourBoard.hasAPlayerWon(newBoard, playerWhoMadeLastMove)) {
+            return 10;
         } else if (emptySpaces.size() == 0) {
             return 0;
         }
-        
+
+        addScoreForSpace(emptySpaces, newBoard, playerWhoIsMoving, scores);
+        return getMaxScore(scores);
+    }
+
+    public void addScoreForSpace(ArrayList<Integer> emptySpaces, ArrayList newBoard, String playerWhoIsMoving, ArrayList<Integer> scores) {
         for (int space : emptySpaces) {
             newBoard.set(space, playerWhoIsMoving);
-            int score = miniMax(newBoard, depth, playerWhoIsMoving);
+            int score = miniMax(newBoard, playerWhoIsMoving);
             scores.add(score);
+            newBoard.set(space, space);
         }
+    }
 
-        if (playerWhoMadeLastMove.equals(aiSymbol)) {
-            int maxScore = Collections.max(scores);
-            scores.clear();
-            return maxScore;
-        }
-        int minScore = Collections.min(scores);
+    public int getMaxScore (ArrayList<Integer> scores) {
+        int maxScore = Collections.max(scores);
         scores.clear();
-        return minScore;
+        return -maxScore;
+    }
+
+    public void easyMode(IBoard board) {
+        for (Object space : board.getSpaces()) {
+            if (space instanceof Integer) {
+                board.updateSpace(space.toString(), this.aiSymbol);
+                break;
+            }
+        }
     }
 
     public void makeMove(IBoard board) {
         if (difficulty.equals("Easy")) {
             easyMode(board);
         }
-
+        if (difficulty.equals("Medium")) {
+            easyMode(board);
+        }
         if (difficulty.equals("Hard")) {
```

Comment:
Why is easy mode separated out but not hard mode? Please see my comment on your last pr for advice on how this section is organized